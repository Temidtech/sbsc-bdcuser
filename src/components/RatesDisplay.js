
import React, { Component } from 'react';
import { Text, View, Image, ActivityIndicator } from 'react-native';
import { RateScreenStyles as styles } from '../styles';
import { Query } from 'react-apollo';
import { sortTodayRates } from '../dataApi';
import { GET_TODAY_RATES } from '../operations';
import { RatesDisplayItem, Whoops } from '.';
import Images from '@assets/images';
import moment from 'moment';

export default class RatesDisplay extends Component {
  render() {
    const formattedDate = moment().format('ddd. MMM D');
    return (
      <View>
        <View style={[styles.row, styles.header]}>
          <View style={styles.date}>
            <Text style={styles.white}>{formattedDate}</Text>
          </View>
          <View style={styles.currency}>
            <Text style={styles.white}>Currency</Text>
          </View>
          <View style={styles.buyRate}>
            <Image source={Images.ngr} style={styles.headerImage} />
            <Text style={styles.white}>Buy Rate</Text>
          </View>
          <View style={styles.sellRate}>
            <Image source={Images.ngr} style={styles.headerImage} />
            <Text style={styles.white}>Sell Rate</Text>
          </View>
        </View>
        <Query query={GET_TODAY_RATES}>
          {({ data, loading, error }) => {
            if (error) {
              const message = error.toString().includes('Network') ?
                'Pls check your internet connection' :
                'An error occured. Retry.';
              return (
                <View style={styles.errorView}>
                  <Whoops message={message}/>
                </View>
              );
            }
            if (loading) {
              return (
                <View style={styles.loaderView}>
                  <ActivityIndicator size="large" />
                </View>
              );
            }
            if (data) {
              const location = this.props.navigation.state.key;
              const transformedData = sortTodayRates(data)[location] || {};
              return (
                <RatesDisplayItem rates={transformedData} />
              );
            }
          }}
        </Query>
      </View>
    );
  }
}
