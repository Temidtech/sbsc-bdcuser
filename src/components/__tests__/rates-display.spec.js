import React from 'react';
import renderer from 'react-test-renderer';
import RatesDisplay from '../RatesDisplay';

describe('<RatesDisplay />', () => {

  it('renders the RatesDisplay component correctly', () => {
    const tree = renderer.create(
      <RatesDisplay />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
