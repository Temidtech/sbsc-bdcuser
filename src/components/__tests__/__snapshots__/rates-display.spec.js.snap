// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RatesDisplay /> renders the RatesDisplay component correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "justifyContent": "space-between",
          "marginTop": 6,
          "paddingLeft": 18,
          "paddingRight": 18,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontFamily": "OpenSans-Italic",
            "fontSize": 14,
            "marginLeft": 4,
          }
        }
      >
        Wed, 29th April
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontFamily": "OpenSans-Italic",
            "fontSize": 14,
            "marginLeft": 4,
          }
        }
      >
        Currency
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "height": 14,
              "marginTop": 2,
              "width": 14,
            },
            Object {
              "backgroundColor": "#FF9E3D",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontFamily": "OpenSans-Italic",
            "fontSize": 14,
            "marginLeft": 4,
          }
        }
      >
        Buy Range
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 4,
              "height": 14,
              "marginTop": 2,
              "width": 14,
            },
            Object {
              "backgroundColor": "#FF0000",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontFamily": "OpenSans-Italic",
            "fontSize": 14,
            "marginLeft": 4,
          }
        }
      >
        Sell Range
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Hello from the other side
  </Text>
</View>
`;
